tests/rados/test_norecovery_flag_functionality.py

```diff
@@ -0,0 +1,81 @@
+"""
+This file contains the  methods to verify the  norecover flag.
+"""
+
+import time
+import traceback
+from datetime import datetime, timedelta
+
+from ceph.ceph_admin import CephAdmin
+from ceph.rados import utils
+from ceph.rados.core_workflows import RadosOrchestrator
+from utility.log import Log
+from utility.utils import method_should_succeed
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+     Test to Verify the norecover flag.
+    Returns:
+        1 -> Fail, 0 -> Pass
+    """
+
+    log.info(run.__doc__)
+    config = kw["config"]
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_obj = RadosOrchestrator(node=cephadm)
+    config = kw["config"]
+    # Creating ec pool
+    replicated_config = config.get("replicated_pool")
+    pool_name = replicated_config["pool_name"]
+
+    try:
+        utils.configure_osd_flag(ceph_cluster, "set", "noscrub")
+        log.info("The noscrub is set on the cluster")
+        utils.configure_osd_flag(ceph_cluster, "set", "nodeep-scrub")
+        log.info("The nodeep-scrub is set on the cluster")
+        utils.configure_osd_flag(ceph_cluster, "set", "nobackfill")
+        log.info("The nobackfill is set on the cluster")
+        utils.configure_osd_flag(ceph_cluster, "set", "norecover")
+        log.info("The norecover is set on the cluster")
+        if not rados_obj.create_pool(name=pool_name, **replicated_config):
+            log.error("Failed to create the  Pool")
+            return 1
+        log.info(f"The pool-{pool_name} is created")
+        # check the ceph df
+        end_time = datetime.now() + timedelta(seconds=3600)
```

Comment:
Before entering the loop, please trigger some autoscaler activity.

Please refer to the bug, to get the area where the fix was provided.

Reply:
The following five scenarios are covered-
Scenario1: Verify that the pg_num should not  increase after setting the norecover flag
Scenario2: Verify that the pg_num should not decrease after setting the norecover flag
Scenario3:Modify the pg number manually and verify that the pg_num should not increase after setting the norecover flag
Scenario4:Modify the pg number manually and verify that the pg_num should not decrease after setting the norecover flag
Scenario5:Checking the recovery of cluster after setting norecover flag